### ISSUE TITLE
Duplicated Version numbers

### ISSUE THREAD

**Issue by Kroseida** [thenativeweb/get-next-version#41]
Version: 2.1.1

Hey.

At the moment I have some problems with your tool.
The correct version of my repository is not recognized.
It always hangs at 0.1.0.

Release Workflow:

```
name: release
on:
  push:
    branches:
    - 'develop'
    
jobs:
  tag:
    name: release
    runs-on: ubuntu-latest

    steps:    
    - name: Clone repository
      uses: actions/checkout@v3
      with:
        fetch-depth: 0
    
    - name: Set up JDK 17
      uses: actions/setup-java@v3
      with:
        java-version: '17'
        distribution: 'temurin'
        cache: maven  
    
    - name: Use Node.js 16.x
      uses: actions/setup-node@v3
      with:
        node-version: 16.x
    
    - name: Get next version
      id: get_next_version
      uses: thenativeweb/get-next-version@2.1.1
    
    - name: Create new release
      if: ${{ steps.get_next_version.outputs.hasNextVersion == 'true' }}
      run: |
        TARGET_VERSION=${{ steps.get_next_version.outputs.version }}
        cd backend
        mvn versions:set -DnewVersion=$TARGET_VERSION
        cd ..
        cd frontend
        npm version $TARGET_VERSION
        cd .. 
        git config --global user.email "actions@github.com"
        git config --global user.name "Github Actions"
        git add backend/pom.xml
        git add frontend/package.json
        git commit -m "Changed version to $TARGET_VERSION" --no-verify
        git tag $TARGET_VERSION
        git push origin $TARGET_VERSION
        
    - name: Build Backend
      if: ${{ steps.get_next_version.outputs.hasNextVersion == 'true' }}
      run: |
        cd backend
        mvn -B package --file pom.xml
        
    - name: Zip Backend
      uses: montudor/action-zip@v1
      with:
        args: zip -qq -r backend.zip backend/target
    
    - name: Build Frontend
      if: ${{ steps.get_next_version.outputs.hasNextVersion == 'true' }}
      run: |
       cd frontend
       npm ci
       npm run build --if-present
        
    - name: Zip Frontend
      uses: montudor/action-zip@v1
      with:
        args: zip -qq -r frontend.zip frontend/dist/spa
        
    - name: Create Release
      id: create_release
      uses: actions/create-release@v1
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      with:
        tag_name: ${{ steps.get_next_version.outputs.version }}
        release_name: ${{ steps.get_next_version.outputs.version }}
        draft: false
        prerelease: true
        
    - name: Upload Backend Release Asset
      uses: actions/upload-release-asset@v1
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      with:
        upload_url: ${{ steps.create_release.outputs.upload_url }}
        asset_path: backend.zip
        asset_name: backend.zip
        asset_content_type: application/zip
     
    - name: Upload Frontend Release Asset
      uses: actions/upload-release-asset@v1
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      with:
        upload_url: ${{ steps.create_release.outputs.upload_url }}
        asset_path: frontend.zip
        asset_name: frontend.zip
        asset_content_type: application/zip
```
![image](https://user-images.githubusercontent.com/18089806/184479157-f44c50ae-ba4d-4028-8504-c538e1f6e712.png)
(The 3 last commits are just for testing)

**Comment by goloroden** [comment]
Hey @Kroseida 👋 

It's a little bit hard to examine the cause of your issue without having access to the repository, but one thing that caught my eye was:

Is the `0.1.0` tag attached to a commit on the `develop` branch or only to a commit of a side-branch?

Even if it was only attached to a commit of a side-branch, I'm not sure whether that's the cause of the issue, but this is one thing you could test: Do things change if you add a tag to the `develop` branch?

**Comment by Kroseida** [comment]
The 0.1.0 tag is attached to the develop branch + one commit (commit to update the version in pom.xml / package.json).

I cleared the git history and removed the commit to update the version. Now it seems to work.
So maybe is tagging with a commit (but without commit into develop) not working?

(In the first try of my github workflow I also pushed the new version into develop. But protecting the branch will not allow this.)
